### PR TITLE
Professor approve student requests

### PR DESCRIPTION
**Pull request by s312121**

### CODE REVIEW

File: server/database.db
Comment (rafalapetina) [inline server/database.db:1]: was there any structural change to the db? If not I would suggest to no include it in the pr

File: server/src/services/proposal.services.js [hunk 1]
```diff
@@ -748,47 +862,49 @@ export const createProposalRequest = async (
...
+          notes: notes,
+          status: "submitted",
         });
       }
     );
   });
 };
 
-export const getProposalRequestsFromDB = async () => {
+export const getProposalRequestsFromDB = async (teacherId = undefined) => {
```
Comment (rafalapetina) [inline server/src/services/proposal.services.js:880]: Is this `teacherId` supposed to be here? If yes why?

File: server/tests/unit/proposal.unit.test.js [hunk 2]
```diff
@@ -986,6 +1017,68 @@ describe('changeStatusProposalRequest', () => {
...
+    };
+
+    await controllers.updateThesisStatus(
+      { params: { proposalId: 115 }, body: { status } },
+      mockResponse
+    );
+
+    expect(mockResponse.status).toHaveBeenCalledWith(400);
+    expect(mockResponse.json).toHaveBeenCalledWith({
+      error: "Uncorrect id",
```
Comment (rafalapetina) [inline server/tests/unit/proposal.unit.test.js:1043]: Here the test says it should return a success message, but it's testing that it returns an error

File: server/tests/unit/proposal.unit.test.js [hunk 3]
```diff
@@ -986,6 +1017,68 @@ describe('changeStatusProposalRequest', () => {
...
+      status: jest.fn().mockReturnThis(),
+      json: jest.fn(),
+    };
+
+    await controllers.updateThesisStatus(
+      { params: { proposalId: 115 }, body: { status } },
+      mockResponse
+    );
+
+    expect(mockResponse.status).toHaveBeenCalledWith(400);
```
Comment (rafalapetina) [inline server/tests/unit/proposal.unit.test.js:1081]: Also here it says that it's returning a 500 code but testing a 400

**Review by rafalapetina — CHANGES_REQUESTED** [review]

File: server/database.db
Comment (s312121) [reply server/database.db:1]: Yes, two columns have been added to the Proposals table (thesis_request_status with 'pending' default and thesis_request_status_date).

File: server/src/services/proposal.services.js (hunk 1 above)
Comment (s312121) [reply server/src/services/proposal.services.js:880]: Thank you for your observation. No it was just for testing. I'll remove it.

**Review by rafalapetina — APPROVED** [review]

**Pull request merged**